mantidproject/mantid
# Use sectors drawn in the instrument viewer for Q1DWeighted

**Description of work.**
This PR makes it possible to use sectors drawn as masks in the instrument viewer as input to define integration regions in Q1DWeighted.

Sectors drawn in the mask tab can be saved using `Save mask as table`, and this table can be passed as argument


**To test:**
- Load a file, open the instrument viewer in the Full 3D, Z- projection, and draw some sectors - ideally some for which you have some idea what the Q1D integration would look like - , without rotating the camera (but translation and zoom are supported)
- Save them to table
- Run Q1DWeighted with the saved table as `ShapeTable` input, and check that the result you have is reasonable.

- In SANSILLIntegration, check that the irrelevant fields (`NumberWedges`, `WedgeAngle` and `WedgeOffset`) are greyed if a table is provided


WIP for #30343 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
